Actions control hook execution expression (#5707)
treeverse/lakeFS: docs/hooks/overview.md

{: .no_toc }

An **Action** is a list of Hooks with the same trigger configuration, i.e. an event will trigger all Hooks under an Action or none at all.
The Hooks under an Action are ordered and so is their execution.

> Before each hook execution the `if` boolean expression is evaluated. The expression can use `success()` which evaluate to true in case of no error was occured while running the hoook's actions. It can also use `failure()` which evaluate to true when previous step failed - can be used to report failures.

Reviewer: ```suggestion
Before each hook execution the `if` boolean expression is evaluated. The expression can use `success()` which evaluate to true in case no error has occurred while running the hook's actions. It can also use `failure()` which evaluates to true if a previous step failed and can be used to report failures as well.
```

> By default, when `if` is empty or omitted, the step will run only if no error occurred (same as evaluate of success function).

Author: very similar to GitHub actions. it evaluates to a check of the last error that is set once a hook fails. It is what we did so far, just broke the processing loop. now we continue to evaluate each step.

> By default, when `if` is empty or omitted, the step will run only if no error occurred (same as evaluate of success function).

Reviewer: Yes but how would you write the condition to continue in case of errors in practice? I think this might be confusing

> By default, when `if` is empty or omitted, the step will run only if no error occurred (same as evaluate of success function).

Author: Maybe I don't understand the question. But in practice you write `if: success()` or nothing in case you like the step to execute if everything green until that point.
In case you like to run always `if: true`.
In case you like to run if something failed `if: failure()`.
